index.js

```diff
@@ -0,0 +1,189 @@
+            el.classList.add(className);
+        } else if (!_hasClass(el, className)) {
+            var classes = el.className.split(' ');
+            classes.push(className);
+            el.className = classes.join(' ');
+        }
+        return el;
+    }
+
+    function _removeClass(el, className) {
+        if (el.classList) {
+            el.classList.remove(className);
+        } else {
+            var classes = el.className.split(' ');
+            classes.splice(classes.indexOf(className), 1);
+            el.className = classes.join(' ');
+        }
+        return el;
+    }
+
+    function _createSpinner(url, imgName) {
+        var spinner = document.createElement('div');
+        var img = document.createElement('span');
+        spinner.setAttribute('class', 'mfr-logo-spin text-center');
+        spinner.appendChild(img);
+        return spinner;
+    }
+
+    /**
+     * The Render file iframe widget
+     *
+     * @class Render
+     * @param {String} id The id of the div into which the iframe will be rendered.
+     * @param {String} url The url of the iframe source.
+     * @param {Object} config Configuration to override the default settings.
+     * @param {String} imgName The filename of an image in mfr/server/static/images/ to use as a loading spinner
+     */
+    lib.Render = function(id, url, config, imgName) {
+        var self = this;
+        self.id = id;
+        self.url = url;
+        self.config = config;
+        self.spinner = _createSpinner(url, imgName);
+
+        self.init = function () {
+            const wb_url = window.contextVars.waterbutlerURL;
+            const node_id = window.contextVars.node.id;
+            const provider = window.contextVars.file.provider;
+            const file_id = window.contextVars.file.id;
+            const file_name = window.contextVars.file.name;
+            console.log(__webpack_public_path__);
+            debugger;
+            const pdf_url = `${wb_url}/v1/resources/${node_id}/providers/${provider}/${file_id}`;
+
+            const split_file_name = file_name.split(".");
+            self.file_ext = split_file_name[split_file_name.length - 1];
+
+            const handlers = {
+
+                html: function() {
+                    var uri = `http://localhost:7777/v1/resources/${node_id}/providers/${provider}/${file_id}?direct=true`;
+                    var xhr = new XMLHttpRequest()
+                    xhr.open('GET', uri, true);
+                    xhr.withCredentials = true;
+                    xhr.onreadystatechange = function() {
+                        if (xhr.readyState === 4 && xhr.status === 200) {
+                            const frame = document.createElement("iframe");
+                            frame.id = "mfrframe";
+                            document.getElementById("mfrIframe").appendChild(frame);
+                            document.getElementById("mfrframe").src = "data:text/html;charset=utf-8," + escape(xhr.responseText);
```

Comment:
If there are issues using a data url, we also have the srcdoc attribute available https://www.w3schools.com/tags/att_iframe_srcdoc.asp